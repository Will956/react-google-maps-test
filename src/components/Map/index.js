import React from 'react';
import { GoogleMap, withGoogleMap, withScriptjs } from 'react-google-maps';

import { defaultZoom, defaultCenter } from '../../config';

const Map = () => (
  <GoogleMap defaultZoom={defaultZoom} defaultCenter={defaultCenter} />
);

export default withScriptjs(withGoogleMap(Map));
