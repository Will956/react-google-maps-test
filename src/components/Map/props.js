import PropTypes from 'prop-types';

export const propTypes = {
  sessions: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      description: PropTypes.string,
      date: PropTypes.string,
      coordinates: PropTypes.shape({
        lat: PropTypes.number,
        lng: PropTypes.number
      }),
      max_participant: PropTypes.number,
      nb_participant: PropTypes.number,
      type: PropTypes.string
    })
  )
};

export const defaultProps = {
  sessions: []
};
