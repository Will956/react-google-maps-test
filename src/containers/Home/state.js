import { connect } from 'react-redux';

import { fetchSessions } from '../../redux/sessions/actions';
import { fetchWeather } from '../../redux/weather/actions';

export const mapStateToProps = state => ({
  sessions: state.sessions,
  weather: state.weather
});

export const mapDispatchToProps = dispatch => ({
  fetchSessions: () => dispatch(fetchSessions()),
  fetchWeather: () => dispatch(fetchWeather())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
);
