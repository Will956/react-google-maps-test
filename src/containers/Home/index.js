import React from 'react';
import classNames from 'classnames';
import { googleMapURL } from '../../config';

import Map from '../../components/Map';

import s from './index.scss';

const Home = () => (
  <div>
    <Map
      googleMapURL={googleMapURL}
      loadingElement={
        <div
          style={{
            height: `100%`
          }}
        />
      }
      containerElement={<div className={classNames(s['map-container'])} />}
      mapElement={<div className={classNames(s.map)} />}
    />
  </div>
);

export default Home;
