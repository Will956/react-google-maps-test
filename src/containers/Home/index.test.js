import React from 'react';
import { shallow } from 'enzyme';
import { googleMapURL } from '../../config';

import Home from './index';
import Map from '../../components/Map';

describe('<Home /> tests', () => {
  it('snapshot: renders default state', () => {
    const wrapper = shallow(<Home />);

    expect(wrapper).toMatchSnapshot();
  });

  it('correctly takes googleMapURL filled in the config', () => {
    const wrapper = shallow(<Home />);

    expect(wrapper.find(Map).props().googleMapURL).toEqual(googleMapURL);
  });
});
