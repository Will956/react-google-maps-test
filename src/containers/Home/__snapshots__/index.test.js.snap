// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> tests snapshot: renders default state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <withScriptjs(withGoogleMap(Map))
        containerElement={
                <div
                        className=""
                />
        }
        googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyDte3SPQoQY9L2tkAPFAJcld-TFdlGleWs&v=3.exp&libraries=geometry,drawing,places"
        loadingElement={
                <div
                        style={
                                Object {
                                        "height": "100%",
                                      }
                        }
                />
        }
        mapElement={
                <div
                        className=""
                />
        }
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "containerElement": <div
          className=""
/>,
        "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyDte3SPQoQY9L2tkAPFAJcld-TFdlGleWs&v=3.exp&libraries=geometry,drawing,places",
        "loadingElement": <div
          style={
                    Object {
                              "height": "100%",
                            }
          }
/>,
        "mapElement": <div
          className=""
/>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <withScriptjs(withGoogleMap(Map))
          containerElement={
                    <div
                              className=""
                    />
          }
          googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyDte3SPQoQY9L2tkAPFAJcld-TFdlGleWs&v=3.exp&libraries=geometry,drawing,places"
          loadingElement={
                    <div
                              style={
                                        Object {
                                                  "height": "100%",
                                                }
                              }
                    />
          }
          mapElement={
                    <div
                              className=""
                    />
          }
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "containerElement": <div
            className=""
/>,
          "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyDte3SPQoQY9L2tkAPFAJcld-TFdlGleWs&v=3.exp&libraries=geometry,drawing,places",
          "loadingElement": <div
            style={
                        Object {
                                    "height": "100%",
                                  }
            }
/>,
          "mapElement": <div
            className=""
/>,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
